Integrate friends

okg00dbye opened the pull request:
Changes
Backend
- Implement Register
- Implement GetProfile, GetFriends and GetFriendRequests
- Implement SendFriendRequest, AcceptFriendRequest, DeclineFriendRequest

Frontend
- Add user store
- Add methods for getFriends, getFriendRequests, send/accept/decline FriendRequests, getMyProfile in user store
- Add user's geolocation in the store, updated from useGeolocation and read in GemMap
- Call getMyProfile upon log in
- Linked add friend + accept/decline friend requests


TODO:
- Correct the SQL [drafting]
- Update seed to have some friend relationships to test
- Link fetched data to frontend UI
- Test this

okg00dbye committed:
Add GetFriends rpc to friend.proto

okg00dbye committed:
Add friendservice

okg00dbye committed:
Add friend repository

okg00dbye committed:
Add friends usecases and repo interface

okg00dbye committed:
Add friends store

okg00dbye committed:
Update GetFriendsResponse to return more info than id

okg00dbye committed:
Add repo for friend requests

sebastiantoh commented on frontend/src/store/modules/friends.ts:
```diff
@@ -0,0 +1,39 @@
+import { RootState } from "@/store";
+import { Module } from "vuex";
+
+export interface FriendsState {
```
Not sure if we should rename this module to User? Then in the user store, we can store:
- friends
- signed in user (upon login, we set the store with the signed in user)
- user's current geolocation

Thoughts?

okg00dbye replied:
makes sense! will try

sebastiantoh commented on frontend/src/store/modules/friends.ts:
```diff
@@ -0,0 +1,39 @@
+
+export interface FriendsState {
+  // List of friend's ids
+  friends: number[];
```
hmm, we probably need to store more than just the IDs. Maybe we need the name as well?

okg00dbye replied:
Yeah oops this part is abit outdated~ I changed it to return username and displayname (and need avatar also but haven't implemented) in the proto and be, will update

okg00dbye committed:
Update UserState in user store (renamed from friends)

okg00dbye committed:
Move Friends and FriendRequests to be under User

okg00dbye committed:
Add GetFriends / GetFriendRequests methods to user.go

okg00dbye committed:
hg fix

okg00dbye committed:
Add getFriendRequests to user store

okg00dbye committed:
Retrieve UserID from context instead

okg00dbye committed:
Update friend.proto

okg00dbye committed:
Add usescases for Send/Accept/Decline friend requests

okg00dbye committed:
implement send friend request

okg00dbye committed:
Implement Accept/Decline FriendRequest

okg00dbye committed:
Register FriendServiceServer

okg00dbye committed:
Update user store with friend request methods

okg00dbye committed:
Misc cleanup

okg00dbye committed:
Define new FriendRelationship struct

Open to name suggestions~

okg00dbye committed:
Update GetFriendRequests to also return timestamp

okg00dbye committed:
Add currPosition into user store

okg00dbye committed:
Fix currentPosition not being set properly

okg00dbye committed:
Remove SayHello rpc from gem.proto

okg00dbye committed:
Move GetFriends/GetFriendRequests from friend to user

okg00dbye committed:
Update UserService

okg00dbye committed:
Register UserClient with the store

okg00dbye committed:
Remove SayHello from gemService and example client

okg00dbye committed:
Implement GetProfile in user service

okg00dbye committed:
Add getMyProfile in user store

okg00dbye committed:
Call getMyProfile upon log in

okg00dbye committed:
Link addFriend button to method

okg00dbye committed:
Link accept/decline friend request buttons to their methods

okg00dbye committed:
Update RPC method names to match service

okg00dbye committed:
Add avatar field to PersonInfo proto

okg00dbye committed:
Update userService to pass avatar

okg00dbye committed:
hg fix

mah bad

ianyong committed:
Separate out declarations of repositories and services

To make it easier to read.

ianyong committed:
Convert snake case to camel case for Go variables

ianyong committed:
Return nil (zero value) when error occurs

ianyong committed:
Add friends and friend_requests tables

ianyong committed:
Fix SQL for userRepository.GetFriends

ianyong committed:
Merge branch 'master' into jl/integrate-friends

okg00dbye committed:
Add RPC for Register

okg00dbye committed:
Add Register usecase

okg00dbye committed:
Add Register method in auth store

okg00dbye committed:
Link RegisterButton

sebastiantoh commented on frontend/src/pages/LoginRegister/RegisterPage.vue:
```diff
@@ -31,6 +32,32 @@
+  methods: {
+    register() {
+      if (this.password !== this.confirmPassword) {
+        // TODO: show some error here.
```
For this we can probably use add a rule / validator using https://youzan.github.io/vant/#/en-US/form#validate-rules instead of manually handling it?

ianyong committed:
Merge branch 'master' into jl/integrate-friends

ianyong committed:
Fix typo in dispatching of register function

ianyong committed:
Make use of useStore

ianyong committed:
Make use of camel case in frontend

ianyong committed:
Add protoUserToUserMapper

ianyong committed:
Hook up SelectFriend to backend

ianyong committed:
Add search by username functionality

ianyong committed:
Rename PersonInfo to User

Rationale:
1) Person and user refer to the same concept, so having fewer terms
   is less confusing.
2) The "Info" suffix does not add any meaning since everything that
   is being passed around RPC calls are information of sorts.

ianyong committed:
Merge FriendService into UserService

This distinction wasn't very meaningful since the responsibilities
were not well defined.

ianyong committed:
Indent user.proto using spaces for consistency

ianyong committed:
Standardise use of google.protobuf.Empty

ianyong committed:
Standardise RPC method signature naming

ianyong committed:
Reorder messages in user.proto for readability

ianyong commented:
I will clean up everything in smaller PRs; scope is getting too big.
